File: backend/app/schemas/job.py
from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel, ConfigDict


class JobBase(BaseModel):
    title: str
    company: str
    location: str
    work_mode: str = "Remote"
    employment_type: str = "Full-time"
    salary_range: Optional[str] = None
    description: str
    requirements: Optional[str] = None
    skills_required: str
    preferred_skills: Optional[str] = None
    education_requirement: Optional[str] = None
    experience_level: Optional[str] = "Entry-level"
    posted_date: Optional[str] = "Recently"
    source: Optional[str] = "Company Careers"
    source_url: Optional[str] = None
    application_url: Optional[str] = None


class JobCreate(JobBase):
    pass


class JobResponse(JobBase):
    model_config = ConfigDict(from_attributes=True)

    id: str
    is_active: bool
    created_at: datetime


class JobMatchResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    job_id: str
    job_title: str
    company: str
    match_score: Optional[int] = None
    match_tier: str  # Excellent Match, Strong Match, Moderate Match, Low Match, Weak Match, Match Unavailable
    matched_skills: List[str]
    missing_skills: List[str]
    explanation: str
    is_profile_insufficient: bool = False
    # Phase 3B Enhanced fields
    strong_matches: List[str] = []
    missing_required_skills: List[str] = []
    missing_preferred_skills: List[str] = []
    weak_skills: List[str] = []
    experience_relevance: int = 0
    education_relevance: int = 0
    profile_completeness: int = 0
    skill_gap_count: int = 0


class SkillGapItem(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    step_number: int
    skill_name: str
    importance: str  # High, Medium, Nice-to-have
    current_proficiency: str
    recommendation: str
    estimated_time: Optional[str] = None
    # Phase 3B Enhanced fields
    why_it_matters: Optional[str] = None
    roadmap_topics: List[str] = []
    suggested_practice: Optional[str] = None


class SkillGapResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    job_id: str
    job_title: str
    match_score: Optional[int] = None
    gaps: List[SkillGapItem]

File: android/app/src/main/java/com/jobpilot/app/data/model/Job.kt
package com.jobpilot.app.data.model

enum class MatchTier {
    EXCEPTIONAL, // 90%+
    STRONG,      // 75%-89%
    MODERATE,    // 60%-74%
    DEVELOPING   // <60%
}

data class Job(
    val id: String,
    val title: String,
    val company: String,
    val companyLogoUrl: String? = null,
    val location: String,
    val workMode: String, // Remote, Hybrid, On-site
    val employmentType: String, // Internship, Full-time
    val stipendOrSalary: String,
    val description: String,
    val postedDaysAgo: Int,
    val requirements: List<String>,
    val matchDetails: JobMatch,
    val source: String? = "Direct",
    val applicationUrl: String? = null
)

data class JobMatch(
    val matchScore: Int? = null,
    val matchTier: MatchTier? = null,
    val strongMatches: List<String> = emptyList(),
    val missingSkills: List<String> = emptyList(),
    val partialMatches: List<String> = emptyList(),
    val whyItMatchesExplanation: String = "",
    val improvementPlan: List<ImprovementStep> = emptyList(),
    val isProfileInsufficient: Boolean = false
)

data class ImprovementStep(
    val stepNumber: Int,
    val title: String,
    val actionDescription: String,
    val estimatedDays: Int = 5,
    val recommendedResource: String = "Interactive Course & Project"
)
